packages/good-design/src/core/web3/SwitchChainActionModal.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { FC, useMemo, useCallback, useEffect } from "react";
+import { useModal } from "../../hooks/useModal";
+import { Box, Text, useColorModeValue } from "native-base";
+import BasePressable from "../buttons/BasePressable";
+import { Title } from "../layout";
+import BackToSchool from "../../assets/images/backtoschool.png";
+import { Image } from "../images";
+import { openLink } from "@gooddollar/web3sdk-v2";
+
+export interface SwitchChainProps {
+  /**  during a active switch request, the modal pops up */
+  switching: boolean;
+}
+
+const SwitchChainActionModal: FC<SwitchChainProps> = ({ switching }: { switching: boolean }) => {
+  const { Modal: SwitchActionModal, showModal, hideModal } = useModal();
+  const textColor = useColorModeValue("goodGrey.500", "white");
+
+  const openNotionTab = useCallback(async () => {
+    const link = "https://www.notion.so/gooddollar/User-Guides-24dd615eb7804792a44057b96b40147d";
+    await openLink(link, "_blank");
+  }, []);
+
+  useEffect(() => {
+    switching ? showModal() : hideModal();
+  }, [switching, showModal, hideModal]);
+
+  const switchChainModalCopy = useMemo(
```

Comment:
remove this, instead

1. header could be defined as `React.memo`'ized component taking single props textColor

```javascript
const Header = memo(({ textColor }) => (
  <Box backgroundColor={"white"}>
     <Title fontSize="xl" mb="2" fontWeight="bold" lineHeight="36px">
       Action Required
     </Title>
     <Text color={textColor} fontFamily="subheading" fontWeight="normal" fontSize="md">
       To complete this action, switch networks in your wallet.
     </Text>
   </Box>
))
```

2. body could be passed as is, just move innerView's style somewhere to default props values (according theming concept) or define it as constant outside component, so it will be the same object each render

```javascript
const innerView = {
             w: "300",
             h: "130px",
             bgColor: "goodWhite.100",
             display: "flex",
             flexDir: "row",
             alignItems: "center",
             justifyContent: "center",
             style: { flexGrow: 1 }
           }

// ...
return (
  <SwitchActionModal  
    header={<Header textColor={textColor} />}
    body={
       (
         <BasePressable
           innerView={innerView}
           onPress={openNotionTab}
         >
        // ...
    }
  />
)
```

3. openNotionTab isn't an dependent variable anymore (see comments above)
4. I suggest to use default theme prop for interview rather than constant

Reply:
don't use 'px' in styles
are you using native base styles?

Reply:
there are only a finite set of native-base default sizes
which could be extended! (but I don't deem it necessary all the time to add a specific value when it maybe is only used once)

But what I have tried to do (I might missed it in some places)

Wherever the 'px' is not defined = native-base size = given size * 4
Wherever the 'px' is defined means we need a specific size based on figma screens

And it's not against the grain of native-base styles
the docs show they use specific sizes with 'px' as well